src/drivers/px4io/px4io.cpp

```diff
@@ -1306,7 +1306,7 @@ PX4IO::io_set_control_state(unsigned group)
 		controls.control[3] = 1.0f;
 	}
 
-	uint16_t regs[_max_actuators];
+	uint16_t regs[_max_controls];
```

Comment:
Can you change it to `uint16_t regs[sizeof(controls.control)/sizeof(controls.control[0])];`, and then make sure `_max_controls <= sizeof(controls.control)/sizeof(controls.control[0])`?
We generally don't want to use dynamically sized arrays on the stack.

Reply:
@bkueng - about uint16_t, I didn't get that. its uint16_t already. 
about the other notes, done

Reply:
That is correct, I did not mean to change that (github just shows it a bit strange). Thanks.

Further down you also need to change this line:
```
return io_reg_set(PX4IO_PAGE_CONTROLS, group * PX4IO_PROTOCOL_MAX_CONTROL_COUNT, regs, _max_controls); 
```
To something like:
```
return io_reg_set(PX4IO_PAGE_CONTROLS, group * PX4IO_PROTOCOL_MAX_CONTROL_COUNT, regs, math::min(_max_controls, sizeof(controls.control)/sizeof(controls.control[0])));
```

Reply:
@bkueng, thanks for the note. 
done